core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -168,23 +172,27 @@ public Schema schema() {
 
   @Override
   public PartitionSpec spec() {
-    if (lazySpec == null) {
+    return specs().get(defaultSpecId);
+  }
+
+  @Override
+  public Map<Integer, PartitionSpec> specs() {
+    if (lazySpecs == null) {
       synchronized (this) {
-        if (lazySpec == null && lazyTable == null) {
+        if (lazySpecs == null && lazyTable == null) {
```

Comment:
Opt: A nested if may be clearer to reader, to avoid the same check (lazySpecs == null) twice?

Reply:
This is actually intentional to avoid any race conditions as the initial check is outside of the synchronized block.

Reply:
@aokolnychyi  Oh, I was not clear, I meant the two case within the synchronized block (was not talking the double check locking).
```
if (lazySpecs == null) && (lazyTable == null)
else if (lazySpecs == null)
```
=> (reduce a bit the number of lines reader has to read)
```
if (lazySpecs == null) {
   if (lazyTable == null)
```

Anyway not a big deal